panel/src/components/Forms/Field/EntriesField.vue

```diff
@@ -0,0 +1,438 @@
+				handle: true,
+				list: this.entries,
+				sort: this.isSortable
+			};
+		},
+		/**
+		 * Returns whether the entries can be sorted
+		 * @returns {boolean}
+		 */
+		isSortable() {
+			if (this.disabled === true) {
+				return false;
+			}
+
+			if (this.entries.length <= 1) {
+				return false;
+			}
+
+			if (this.sortable === false) {
+				return false;
+			}
+
+			return true;
+		},
+		/**
+		 * Returns if new entries can be added
+		 * @returns {boolean}
+		 */
+		more() {
+			if (this.disabled === true) {
+				return false;
+			}
+
+			if (this.max && this.entries.length >= this.max) {
+				return false;
+			}
+
+			return true;
+		},
+		options() {
+			return [
+				{
+					disabled: this.entries.length === 0 || this.disabled,
+					icon: "template",
+					text: this.$t("copy.all"),
+					click: this.copyAll
+				},
+				"-",
+				{
+					disabled: this.entries.length === 0 || this.disabled,
+					icon: "trash",
+					text: this.$t("delete.all"),
+					click: this.removeAll
+				}
+			];
+		},
+		values() {
+			return this.entries.map((entry) => entry.value);
+		}
+	},
+	watch: {
+		value: {
+			handler(entries) {
+				// no need to add ids again if the values are the same
+				if (entries === this.values) {
+					return;
+				}
+
+				this.entries = entries.map((value) => {
+					return {
+						id: this.$helper.uuid(),
+						value
+					};
+				});
+			},
+			immediate: true
+		}
+	},
+	methods: {
+		add(index = null, value = null) {
+			if (this.more === false || this.disabled === true) {
+				return;
+			}
+
+			const entry = {
+				id: this.$helper.uuid(),
+				value: value ?? ""
+			};
+
+			if (index !== null) {
+				this.entries.splice(index, 0, entry);
+			} else {
+				this.entries.push(entry);
+				index = this.entries.length - 1;
+			}
+
+			this.save();
+
+			this.$nextTick(() => {
+				this.focus(index);
+			});
+		},
+		copyAll() {
+			const copy = this.values.map((value) => "- " + value).join("\n");
+
+			this.$helper.clipboard.write(copy);
+			this.$panel.notification.success(this.$t("copy.success"));
+		},
+		duplicate(index) {
+			if (
+				this.more === false ||
+				this.disabled === true ||
+				this.entries[index] === undefined
+			) {
+				return;
+			}
+
+			const duplicate = {
+				...this.entries[index],
+				id: this.$helper.uuid()
+			};
+
+			this.entries.splice(index + 1, 0, duplicate);
+			this.save();
+
+			this.$nextTick(() => {
+				this.focus(index + 1);
+			});
+		},
+		focus(index, on = "input") {
+			this.$refs["entry-" + index + "-" + on]?.[0]?.focus?.();
+		},
+		onInput(index, value) {
+			this.entries[index].value = value;
+			this.save();
+		},
+		onPaste(input) {
+			if (input instanceof ClipboardEvent) {
+				input = this.$helper.clipboard.read(input, true);
+			}
+
+			const entries = input
+				.split(/\r?\n/)
+				.filter(entry => entry.trim().length > 0)
+				.map(entry => entry.startsWith("- ") ? entry.slice(2) : entry);
+
+			entries.forEach(entry => this.add(null, entry));
```

Comment:
```suggestion
			for (const entry of entries) { 
				this.add(null, entry);
			}
```

Reply:
Sometimes I write the short version, thinking @distantnative would write shorter, and then we move on to the long version. When I write the long version, and then we move on to the short version. What's the logic behind this? Can someone explain it to me? 🙈😄

Reply:
:D Sorry, it might be confusing as it's not always about shorter/longer.

In this case, we have moved away from using too many methods where the following code is nested in a callback. Which is why we dropped most uses of `.forEach()` completely and went with `for ()` loops instead. But it's a total `Nitpick` - you don't need to accept it :)